src/IsNaughty.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * isWordNaughty
+ * @param shuffledWord    A shuffled word to evaluate as naughty or not.
+ * @param bannedWords     A set of banned words that will flag any substring
+ *                        of the input data as "naughty".
+ * @returns               True if the word is found to be naughty, and false
+ *                        otherwise.
+ */
+export function isWordNaughty(
+  shuffledWord: string,
+  bannedWords: Set<string>
```

Comment:
since you won't modify the set

```suggestion
  bannedWords: ReadonlySet<string>
```